pkg/controller/kibanaassociation/association_controller_test.go

```diff
@@ -290,6 +291,80 @@ func Test_deleteOrphanedResources(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "No more es ref in Kibana, orphan user for previous es ref in a different namespace still exist",
+			kibana: kbv1.Kibana{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "kibana-foo",
+					Namespace: "ns2",
+					UID:       kibanaFixtureUID,
+				},
+			},
+			es: esv1.Elasticsearch{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "es-foo",
+					Namespace: "ns1",
+					UID:       "f8d564d9-885e-11e9-896d-08002703f062",
+				},
+			},
+			initialObjects: []runtime.Object{
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kibana-foo-kibana-user",
+						Namespace: "ns2",
+						Labels: map[string]string{
+							AssociationLabelName:      "kibana-foo",
+							AssociationLabelNamespace: "ns2",
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							ownerRefFixture,
+						},
+					},
+				},
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "ns2-kibana-foo-kibana-user",
+						Namespace: "ns1",
+						Labels: map[string]string{
+							label.ClusterNameLabelName: "es-foo",
+							AssociationLabelName:       "kibana-foo",
+							AssociationLabelNamespace:  "ns2",
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							esRefFixture,
+						},
+					},
+				},
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kibana-foo-kb-es-ca",
+						Namespace: "ns2",
+						Labels: map[string]string{
+							AssociationLabelName: "kibana-foo",
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							ownerRefFixture,
+						},
+					},
+				},
+			},
+			postCondition: func(c k8s.Client) {
+				// This works even without labels because mock client currently ignores labels
```

Comment:
I thought the fake client considers labels in `.List()` now?

Reply:
Maybe, I didn't try to refactor the existing code so much. But I'll try to have a look.